tests/test_runner.py

```diff
@@ -0,0 +1,260 @@
+}
+
+
+class TestRunner(unittest.TestCase):
+    """Test cases for runner module"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up test fixtures"""
+        with open('{}/assets/test.xml'.format(SCRIPT_PATH), 'r') as fileh:
+            cls.test_xml = fileh.read()
+
+    def test_getrunner(self):
+        """Ensure getrunner() can create a runner subclass"""
+        result = runner.getrunner('beaker', {'jobtemplate': 'test'})
+        self.assertIsInstance(result, runner.BeakerRunner)
+
+    def test_invalid_getrunner(self):
+        """Ensure getrunner() throws an exception for an invalid runner"""
+        bad_runner = 'pizza'
+        with self.assertRaises(Exception) as context:
+            runner.getrunner(bad_runner, {'jobtemplate': 'test'})
+
+        self.assertTrue(
+            "Unknown runner type: {}".format(bad_runner)
+            in context.exception
+        )
+
+    def test_getxml(self):
+        """Ensure BeakerRunner.getxml() returns xml"""
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getxml({})
+        self.assertEqual(result, self.test_xml)
+
+    def test_getxml_replace(self):
+        """Ensure BeakerRunner.getxml() returns xml with replacements"""
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getxml({'TEST': 'replaced text'})
+        expected_xml = self.test_xml.replace("##TEST##", "replaced text")
+        self.assertEqual(result, expected_xml)
+
+    @mock.patch('subprocess.Popen')
+    def test_getresultstree(self, mock_popen):
+        """Ensure getresultstree() works"""
+        test_xml = "<xml><test>TEST</test></xml>"
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (test_xml, '')
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getresultstree(jobid=0)
+        self.assertEqual(next(x.text for x in result.iter('test')), 'TEST')
+
+    @mock.patch('subprocess.Popen')
+    def test_dumpjunitresults(self, mock_popen):
+        """Ensure dumpjunitresults() works"""
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (
+            'stdout',
+            'stderr'
+        )
+
+        junit_dir = tempfile.mkdtemp()
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        myrunner.dumpjunitresults("J:00001", junit_dir)
+        expected_file = "{}/j_00001.xml".format(junit_dir)
+        self.assertTrue(os.path.exists(expected_file))
+
+        # Clean up
+        os.unlink(expected_file)
+        os.rmdir(junit_dir)
+
+    @mock.patch('skt.runner.BeakerRunner.getresultstree')
+    def test_getconsolelog(self, mock_getresultstree):
+        """Ensure getconsolelog() works"""
+        # Mock up a beaker XML reply
+        beaker_xml = '{}/assets/beaker_results.xml'.format(SCRIPT_PATH)
+        with open(beaker_xml, 'r') as fileh:
+            mocked_xml = fileh.read()
```

Comment:
You can use misc.get_asset_content to avoid writing repeated code

Reply:
Fixed.